microsite/src/main/docs/guides/accessing-metadata.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: docs
+title: Accessing metadata on services
+section: guides
+permalink: /guides/accessing-metadata
+---
+
+# Context on services
+
+Mu provides way to create contexts available in the client and/or in the server. Specifically, it provides the following features.
+
+## Client
+
+* For every RPC call, you need to create an initial context that will be passed to the client
+* The client will have the ability to operate and transform that context, that will be sent to the server in the headers.
+
+## Server
+
+* The server will have the ability to extract the context information from the request headers and use them.
+
+## How to use
+
+We'll assume the following service definition:
+
+```scala mdoc:silent
+import higherkindness.mu.rpc.protocol._
+
+case class HelloRequest(name: String)
+case class HelloResponse(greeting: String)
+
+@service(Protobuf, namespace = Some("com.foo"))
+trait MyService[F[_]] {
+
+  def SayHello(req: HelloRequest): F[HelloResponse]
+
+}
+```
+
+Let's look at how to enable the context on the client side first.
+
+### Client side
+
+Ordinarily, if you don't want to use this feature, you would create a cats-effect
+`Resource` of an RPC client using the macro-generated `MyService.client` method:
+
+```scala mdoc:silent
+import cats.effect._
+import higherkindness.mu.rpc.{ChannelFor, ChannelForAddress}
+
+object OrdinaryClientApp extends IOApp {
+
+  val channelFor: ChannelFor = ChannelForAddress("localhost", 8080)
+
+  val clientRes: Resource[IO, MyService[IO]] =
+    MyService.client[IO](channelFor)
+
+  def run(args: List[String]): IO[ExitCode] =
+    clientRes.use { client =>
+      for {
+        resp <- client.SayHello(HelloRequest("Chris"))
+        _    <- IO(println(s"Response: $resp"))
+      } yield (ExitCode.Success)
+    }
+}
+```
+
+To obtain a client with the context available, use `MyService.contextClient[F, C]` instead of
+`MyService.client`.
+
+This returns a `MyService[Kleisli[F, C, *]]`, i.e. a client which takes
+an arbitrary `C` as input and returns a response inside the `F` effect.
+
+This method requires an implicit instance in the scope. Concretely a `ClientContext[F, C]`
+
+```scala
```

Comment:
question: could we turn this into a mdoc block?

Reply:
This is already defined in the scope. Do you know if there's a way to embeed scala sources?

Reply:
Maybe you could use `mdoc:nest`? https://scalameta.org/mdoc/docs/modifiers.html#nest

Reply:
I tried, but that doesn't stop that code from providing new definitions to the scope. Honestly, I don't think this needs to be checked, it doesn't represent snippets that the users need to copy, just samples about how  the current code looks